pkg/backup/s3.go

```diff
@@ -33,6 +33,14 @@ type s3Backend struct {
 	dir string
 }
 
+type S3Backend struct {
+	s3Backend
+}
+
+func NewS3Backend(S3 *s3.S3, dir string) *S3Backend {
```

Comment:
How about we make the `backend` interface public?
https://github.com/coreos/etcd-operator/blob/master/pkg/backup/backend.go#L19
And keep `s3Backend` private. Then `NewS3Backend()` would just be:
```

func NewS3Backend(S3 *s3.S3, dir string) Backend {
    return &s3Backend{S3, dir}
}
```
@fanminshi Let me know what you think.

Reply:
@hasbro17 that's my plan in this https://github.com/coreos/etcd-operator/pull/1487.